pkg/controller/job/job_controller.go

```diff
@@ -1429,6 +1455,22 @@ func (jm *Controller) recordJobFinished(job *batch.Job, finishedCond *batch.JobC
 	return true
 }
 
+func (jm *Controller) recordJobSucceeded(job *batch.Job) {
+	successCriteriaMet := hasSuccessCriteriaMetCondition(job)
+	if successCriteriaMet == nil {
+		return
+	}
+	var metricReason string
+	switch successCriteriaMet.Reason {
+	case batch.JobReasonSuccessPolicy:
+		metricReason = "JobSuccessPolicy"
+	case batch.JobReasonCompletionsReached:
+		metricReason = "Completions"
+	}
+	jm.recorder.Event(job, v1.EventTypeNormal, "SuccessCriteriaMet", "Job met success criteria")
```

Comment:
_I would prefer to keep this symmetric with FailureTarget. Either do for both, or for none. Since it isn't done for FailureTarget I think we are ok with that._

originally posted by @mimowo https://github.com/kubernetes/kubernetes/pull/126064#discussion_r1677375202

Reply:
That makes sense.
Let's drop this event recording from here.